srai/loaders/osm_loaders/osm_pbf_loader.py

```diff
@@ -105,29 +109,20 @@ def load(
 
         area_wgs84 = self._prepare_area_gdf(area)
 
-        downloaded_pbf_files: Mapping[Hashable, Sequence[Union[str, Path]]]
+        downloaded_pbf_files: Sequence[Union[str, os.PathLike[str]]]
         if self.pbf_file is not None:
-            downloaded_pbf_files = {Path(self.pbf_file).name: [self.pbf_file]}
+            downloaded_pbf_files = [self.pbf_file]
         else:
             downloaded_pbf_files = PbfFileDownloader(
                 download_source=self.download_source,
                 download_directory=self.download_directory,
                 switch_to_geofabrik_on_error=self.switch_to_geofabrik_on_error,
             ).download_pbf_files_for_regions_gdf(regions_gdf=area_wgs84)
 
-        merged_tags = self._merge_osm_tags_filter(tags)
-
-        pbf_handler = PbfFileHandler(tags=merged_tags)
-
-        results = []
-        for region_id, pbf_files in downloaded_pbf_files.items():
-            features_gdf = pbf_handler.get_features_gdf(
-                file_paths=pbf_files, region_id=str(region_id)
-            )
-            matching_features_ids = features_gdf.sjoin(area_wgs84).index
-            results.append(features_gdf.loc[matching_features_ids])
+        pbf_handler = PbfFileHandler(tags_filter=tags, geometry_filter=area_wgs84.unary_union)
```

Comment:
This whole block seems to be duplicated in here and `load_to_geoparquet`. Maybe extract it into a separate function?

Reply:
You're right, I've simplified the code.